### ISSUE TITLE
Add support for recording location history data in the Publisher

### ISSUE THREAD

**Issue by KacperKluka** [ably/ably-asset-tracking-swift#111]
On iOS, there's no support for recording the trip data. You can only replay the data that was recorded on Android. We should be able to collect the location history data on iOS too.

┆Issue is synchronized with this [Jira Story](https://ably.atlassian.net/browse/SDK-737) by [Unito](https://www.unito.io)

**Comment by jamienewcomb** [comment]
low priority for now

**Comment by KacperKluka** [comment]
We got a response from Mapbox about this issue:
> NavSDK for iOS also has [HistoryRecording](https://docs.mapbox.com/ios/navigation/api/2.6.0/Protocols/HistoryRecording.html) and [ReplayLocationManager](https://docs.mapbox.com/ios/navigation/api/2.6.0/Classes/ReplayLocationManager.html), but we don't have guides for those yet. This [code](https://github.com/mapbox/mapbox-navigation-ios/blob/main/Example/ViewController.swift#L491) includes a use case of HistoryRecording in free drive, I hope this helps.

**Comment by lawrence-forooghian** [comment]
This is for debug purposes only.

Mapbox SDK provides a path to a file that it generates (which we provide to user), and internally we also parse it and simplify it (for file size purposes).

Work involved (upon publisher being stopped):

1. get file from Mapbox
2. expose file in API
3. parse file and expose list to the user

The outputs from this can be seen in the lines `             locationHistoryListener?.invoke(LocationHistoryData(historyEvents.toGeoJsonMessages()))` and `rawHistoryCallback?.invoke(historyDataFilepath)` in the `close()` method of `Mapbox.kt`.

**Comment by lawrence-forooghian** [comment]
This functionality is now ready in a Mapbox SDK beta, so we can start developing this: https://github.com/mapbox-collab/Ably-DH-Collab/issues/6#issuecomment-1288557630

**Comment by lawrence-forooghian** [comment]
This is currently blocked pending a reply from Mapbox about why the `HistoryReader` API is returning an empty list of events: https://github.com/mapbox-collab/Ably-DH-Collab/issues/6#issuecomment-1308838907

**Comment by lawrence-forooghian** [comment]
This is now unblocked.
